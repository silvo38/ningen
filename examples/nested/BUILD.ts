#!/usr/bin/env -S deno run --allow-read --allow-write
// The shebang above lets you execute this BUILD.ts file directly.

// Import nested BUILD.ts files for side-effects.
import "./subdir/BUILD.ts";

// In your own code, use:
// import { init } from "https://deno.land/x/ningen@0.0.5/mod.ts";
import { init } from "../../mod.ts";

// Initialises Ningen and retrieves the functions for defining rules and build
// targets.
const ng = init(import.meta.url);

// Write the build.ninja file. Can override the output file if you like.
ng.generate({
  inputs: ng.glob("**/*.ts"),
});
