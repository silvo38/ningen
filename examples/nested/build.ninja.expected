rule append
  command = ./tools/append.sh $in $out
  description = Appending text

rule ningen
  command = ./BUILD.ts
  description = Regenerating ninja file
  generator = 1

build subdir/foo.out: append subdir/foo.txt | tools/append.sh

build build.ninja: ningen  | BUILD.ts
